tests/install_upgrade_operators/network_policy/test_network_policy_components.py

```diff
@@ -0,0 +1,289 @@
+
+
+@pytest.fixture
+def ingress_connectivity_baseline(
+    admin_client, hco_namespace, deployed_client_pod, service_ip, service_port, network_policy_match_labels
+):
+    """
+    Test preparation for ingress tests: Apply allow-all NetworkPolicy, verify traffic works, then remove it.
+    This ensures that without restrictive policies, the connection should succeed.
+    """
+    with AllowAllNetworkPolicy(
+        name="test-allow-all-traffic",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        match_labels=network_policy_match_labels,
+    ):
+        try:
+            deployed_client_pod.execute(
+                command=shlex.split(f"curl -sS -k --connect-timeout 5 https://{service_ip}:{service_port}")
+            )
+        except Exception as e:
+            pytest.fail(
+                f"Ingress connectivity test preparation FAILED: With allow-all NetworkPolicy, "
+                f"connection from test client pod to the CNV service should succeed but got error: {e}"
+            )
+    yield {
+        "deployed_client_pod": deployed_client_pod,
+        "service_ip": service_ip,
+        "service_port": service_port,
+    }
+
+
+@pytest.fixture
+def egress_connectivity_baseline(
+    admin_client, hco_namespace, pods_by_network_policy, network_policy_match_labels, deployed_server_service
+):
+    """
+    Test preparation for egress tests: Apply allow-all NetworkPolicy, verify traffic works, then remove it.
+    This ensures that without restrictive policies, the connection should succeed.
+    """
+    with AllowAllNetworkPolicy(
+        name="test-allow-all-traffic",
+        namespace=hco_namespace.name,
+        client=admin_client,
+        match_labels=network_policy_match_labels,
+    ):
+        pod_name = pods_by_network_policy[0]
+        service_ip = deployed_server_service.instance.spec.get("clusterIP")
+        if ":" in service_ip:
+            pytest.fail(f"This test does not support sending traffic to IPv6: {service_ip}")
+
+        # Get Pod object to use execute() method
+        test_pod = Pod(client=admin_client, namespace=hco_namespace.name, name=pod_name)
+        try:
+            test_pod.execute(
+                command=shlex.split(f"curl -sS --connect-timeout 5 http://{service_ip}:{TEST_SERVER_PORT}")
+            )
+        except Exception as e:
+            pytest.fail(
+                f"Egress connectivity test preparation FAILED: With allow-all NetworkPolicy, "
+                f"connection from {pod_name} to test server pod should succeed but got error: {e}"
+            )
+
+    yield {
+        "pods_by_network_policy": pods_by_network_policy,
+        "hco_namespace": hco_namespace,
+        "server_service_ip": service_ip,
+    }
+
+
+@pytest.mark.polarion("CNV-12305")
+@pytest.mark.parametrize("network_policy_name", [KUBEVIRT_CONSOLE_PLUGIN_NP])
+def test_console_plugin_network_policy_blocks_ingress_unauthorized_traffic(ingress_connectivity_baseline):
```

Comment:
using ingress_connectivity_baseline to return all resources makes it very hard to understand what is done here.
please call the required fixtures directly